### DISCUSSION TITLE
IPv6 Probing

### DISCUSSION THREAD

**Discussion by shenliehuozhi** [projectdiscovery/httpx#260] [Q&A] [answered]
it seems that httpx can not support ipv6?

**Comment by ehsandeep** [comment]
@shenliehuozhi do you have an example for a host pointing to the only **IPV6** with a web server running?

**Reply by itnsec** [reply]
Hi, 
Thanks for your great tools !  

By design a router host in general one ipv4, and many ipv6 (/56 on my targets).
Also, firewall are generaly better managed in ipv4 than ipv6.

So i can identify several equipements (routers, sw, servers) exposing only ipv6 on internet, (often by mistake)

ipv6 support in projectdiscovery tools would be highly appreciated.

**Reply by seeyarh** [reply]
Some organizations are moving towards an ipv6 only internal network
https://www.arin.net/blog/2019/04/03/microsoft-works-toward-ipv6-only-single-stack-network/

**Comment by seeyarh** [comment, marked as answer]
I've put out PRs for fastdialer and retryabledns to support IPv6
https://github.com/projectdiscovery/fastdialer/pull/32
https://github.com/projectdiscovery/retryabledns/pull/44

I don't think any changes are required to httpx. Using the `-probe-all-ips` flag on a domain that has A and AAAA records results in scans using both IPv4 and IPv6
